driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/webdrivermanager/VisualEventDriverListener.java

```diff
@@ -23,158 +23,43 @@
 
 import eu.tsystems.mms.tic.testframework.common.Testerra;
 import eu.tsystems.mms.tic.testframework.pageobjects.UiElementHighlighter;
-import org.openqa.selenium.By;
-import org.openqa.selenium.OutputType;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
-import org.openqa.selenium.support.events.WebDriverEventListener;
+import org.openqa.selenium.support.events.WebDriverListener;
 
 import java.awt.Color;
 
 /**
  * Highlights some WebDriver events in Demo Mode
  */
-public class VisualEventDriverListener implements WebDriverEventListener {
+public class VisualEventDriverListener implements WebDriverListener {
 
-    UiElementHighlighter elementHighlighter = Testerra.getInjector().getInstance(UiElementHighlighter.class);
+    private UiElementHighlighter elementHighlighter = Testerra.getInjector().getInstance(UiElementHighlighter.class);
 
-    @Override
-    public void beforeAlertAccept(WebDriver webDriver) {
-
-    }
-
-    @Override
-    public void afterAlertAccept(WebDriver webDriver) {
-
-    }
-
-    @Override
-    public void afterAlertDismiss(WebDriver webDriver) {
-
-    }
-
-    @Override
-    public void beforeAlertDismiss(WebDriver webDriver) {
-
-    }
-
-    @Override
-    public void beforeNavigateTo(String s, WebDriver webDriver) {
-
-    }
-
-    @Override
-    public void afterNavigateTo(String s, WebDriver webDriver) {
-    }
-
-    @Override
-    public void beforeNavigateBack(WebDriver webDriver) {
-
-    }
-
-    @Override
-    public void afterNavigateBack(WebDriver webDriver) {
-
-    }
-
-    @Override
-    public void beforeNavigateForward(WebDriver webDriver) {
+    // Current driver is additional needed because WebDriverListener methods of before/after element actions only get WebElement, but no current driver.
+    public WebDriver driver;
 
-    }
+    private boolean isDemo = Testerra.Properties.DEMO_MODE.asBool();
```

Comment:
This is a bad practice, because it prevents overriding Properties during runtime.